### ISSUE TITLE
Syntax error parsing field initializer with `ref` expression

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#64720]
Compile:
```csharp
ref struct R
{
    ref int _f = ref F(); // error CS1525: Invalid expression term 'ref'

    public R()
    {
    }

    static ref int F() => throw null;
}
```
Expected: no errors

Actual:
```
(3,18): error CS1525: Invalid expression term 'ref'
```

See related https://github.com/dotnet/roslyn/issues/64725.
